Sources/ArcGISToolkit/Common/AttachmentRenameAlert.swift

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+/// The contents of an alert to rename an attachment.
+struct AttachmentRenameAlert: View {
+    /// The model for the attachment the user has requested be renamed.
+    let attachmentModel: AttachmentModel
+    
+    /// The action to perform when the attachment is renamed.
+    let onRename: (AttachmentModel, String) async throws -> Void
+    
+    /// The attachment's file extension.
+    @State private var fileExtension: String?
+    
+    /// The attachment's new name.
+    @State private var newName = ""
+    
+    var body: some View {
```

Comment:
I'm not sure what's happening, but If I change the name of an attachment, then attempt to change the name of another attachment, the $newName still shows the name of the previous attachment, not the current one.

Also, since there are no `async` methods in the `.task` below, should it be `.onAppear()`? Although if I set a breakpoint there, the second time through it never gets hit, leading me to believe it's not being recreated (or some off thing(.

Reply:
Nice catch. It looks like `AttachmentRenameAlert` was being re-initialized but the task was only running once - same for `onAppear()`.

I found using `task(id: attachmentModel)` would work but then we also need to reset the state properties. I think that all makes the new view more trouble than its worth so I've removed it alltogether.